tests/js/server/aql/aql-queries-simple.js

```diff
@@ -1335,6 +1335,18 @@ function ahuacatlQuerySimpleTestSuite () {
       });
     },
 
+    testLargeQuery : function() {
+      let q = "";
+      const cnt = 1000;
+      for (let i = 0; i < cnt; ++i) {
+        q += `LET v${i} = NOOPT(1)\n`;
+      }
+      q += "RETURN v0";
+      for (let i = 1; i < cnt; ++i) {
+        q += ` + v${1}`;
+      }
+      assertEqual([cnt], AQL_EXECUTE(q, {}, {optimizer: {rules: ['-all']}}).json);
```

Comment:
Can we add another Assertion here that the resulting plan has >1000 Blocks.
And best if you could assert that the Callstack split has happened.
For some systems 1000 Blocks my be still okay without splitting if your StackSize is high enough (or if for some reason we can reduce the stacksize in AQL later on)

Reply:
I intentionally did not add any assertion about callstack splitting. This test case is meant to test that large queries can be executed successfully without caring _how_ this is achieved. Increasing the number of blocks beyond 1000 causes crashes on my windows machine, because `ExecutionNode::walk` is also implemented recursively which results in a stackoverflow. This will be fixed in a separate PR, at which point we can increase the number of blocks in this test.

Reply:
okay. :+1: